apps/verify.c

```diff
@@ -174,6 +175,7 @@ int verify_main(int argc, char **argv)
             break;
         case OPT_SM2HEXID:
             /* try to parse the input as hex string first */
+            sm2_free = 1;
```

Comment:
Instead of using a flag, instead set sm2_id to NULL after this expression and then you can simply unconditionally free.

Reply:
after which expression?

Reply:
On line 183, following the `}` of the `if (sm2_id == NULL) {`

Reply:
I don't quite get the point, I think the conditionally free action at the `end` section is protecting the `sm2_id` set by `OPT_SM2ID` from being freed, which should not be freed in any case.

Reply:
Sorry disregard my comment, I was misreading the code.

Reply:
Okay